koku/masu/util/common.py

```diff
@@ -325,52 +326,74 @@ def batch(iterable, start=0, stop=None, _slice=1):
         yield res
 
 
-def create_enabled_keys(schema, enabled_keys_model, enabled_keys):
+def create_enabled_tags(schema, enabled_tags, provider_type, enabled_value):
     """
-    Creates enabled key records.
-    """
-
-    ctx = {"schema": schema, "model": enabled_keys_model._meta.model_name, "enabled_keys": enabled_keys}
+    Creates enabled tags.
 
-    if not enabled_keys:
-        LOG.info(log_json(msg="no enabled keys found", context=ctx))
+    Args:
+      schema (str)
+      enabled_tags (set)
+      provider_type (str)
+      enabled_value (boolean)
+    """
+    enabled_tag_count = 0
+    ctx = {"schema": schema, "enabled_tags": enabled_tags}
+    LOG.info(log_json(msg="checking tags should be enabled", context=ctx))
+    if not enabled_tags:
+        LOG.info(log_json(msg="skipping tag enablement no tags found", context=ctx))
         return
 
-    LOG.info(log_json(msg="creating enabled key records", context=ctx))
-    changed = False
-
     with schema_context(schema):
-        new_keys = list(set(enabled_keys) - {k for k in enabled_keys_model.objects.values_list("key", flat=True)})
-        if new_keys:
-            changed = True
-            # Processing in batches for increased efficiency
-            for batch_num, new_batch in enumerate(batch(new_keys, _slice=500)):
-                batch_size = len(new_batch)
-                LOG.info(
-                    log_json(msg="create batch", batch_number=(batch_num + 1), batch_size=batch_size, context=ctx)
-                )
-                for ix in range(batch_size):
-                    new_batch[ix] = enabled_keys_model(key=new_batch[ix])
-                enabled_keys_model.objects.bulk_create(new_batch, ignore_conflicts=True)
-    if not changed:
-        LOG.info(log_json(msg="no enabled keys added", context=ctx))
-
-    return changed
-
+        new_tags = list(
+            set(enabled_tags)
+            - {k for k in EnabledTagKeys.objects.filter(provider_type=provider_type).values_list("key", flat=True)}
+        )
```

Comment:
You can save the creation of an additional set object by using a method and a generator expression.
```suggestion
        new_tags = list(
            set(enabled_tags).difference(
                k for k in EnabledTagKeys.objects.filter(provider_type=provider_type).values_list("key", flat=True)
        )
```

Reply:
You could also leave this as a `set` since it gets cast to a list in `batch()`